# Introduction
Microorganisms, the unseen majority, play an important role in environmental and human health.
Globally there are an estimated $4$ to $6 \times 10^{30}$ prokaryotic cells on earth [@whitman1998prokaryotes; @kallmeyer2012global], which drive processes such as the carbon and nitrogen cycle [@whitman1998prokaryotes].
In the human body bacterial cells are as abundant as human cells [@sender2016revised] and aid in fundamental processes such as digestion [@cho2012human; @young2017role].
The human gut microbiome has been linked to numerous diseases ranging from inflammatory bowel disease to autism [@cho2012human; @strati2017new].
Biotherapeutics, probiotics developed to treat disease, have the potential to revolutionize medicine and treat microbiome-linked diseases [@young2017role].
For example fecal microbiome transplants have been successfully used to treat recalcitrant _Clostridium dificile_ infections [@rao2015fecal].


Recent advances in DNA sequencing technology have changed how we measure and thus study microbiomes.
The two of the most commonly used methods to characterize microbial communities are 16S rRNA marker-gene surveys and shotgun metagenomics [@kuczynski2012experimental].
Shotgun metagenomics is the random sequencing of all genomic material in a sample.
16S rRNA marker-gene surveys use targeted sequencing to characterize  community taxonomic composition.
There are benefits and disadvantage to both of these methods [@clooney2016comparing].
Shotgun sequencing is useful for functional information and strain-level analysis whereas marker-gene sequencing provides a more complete view of the community composition.
Additionally, shotgun sequencing is significantly more expensive than marker-gene surveys and is therefore cost prohibitive for large cohort and global studies, such as the human microbiome project [@turnbaugh2007human] and  earth microbiome project [@thompson2017communal].
The focus of this dissertation is assessing 16S rRNA marker-gene survey methods.
However, the apporaches presented are applicable to assessing shotgun metagenomic methods as well.


## 16S rRNA Marker-Gene Surveys
Marker-gene survey data is used to characterize both organismal and community level differences [@goodrich2014conducting].
Differential abundance testing is a type of organismal level analysis used to identify organisms associated with specific treatment conditions, for example a pathogen associated with a disease state.
Ecological diversity metrics are used to characterize microbial community richness and evenness within individual samples (alpha diversity) and sample pairwise similarity (beta diversity).


Marker gene survey data is collected through a complex multi-step measurement process [@goodrich2014conducting].
The measurement process consists of numerous laboratory and computational steps.
The laboratory steps include DNA extraction, PCR, library prep, and sequencing.
The computational steps include pre-processing, feature inference, feature annotation, and normalization.
See Sections \@ref(measurement-process) and \@ref(feature-inference-and-normalization) for a detailed description of the measurement processes.
Limitations in our understanding of the measurement process impede microbiome research.


## Measurement Assessment
Measurement assessment is used to characterize and validate the measurement process and as a result advance microbiome research.
Results from marker-gene surveys indicate a potential connection between obesity and the gut microbiome [@ley2006microbial].
However, a meta-analysis combining data from multiple studies only found a weak association between obesity and the gut microbiome [@sze2016looking].
Given a well characterized measurement process we can better evaluate study conclusions.
In turn, reducing the identification of spurious associations, such as the ones identified by the obesity studies.


A key component of measurement assessment is data with an expected value.
Common sources of data used to assess marker-gene surveys includes, computer simulated data (_in silco_), mixtures of DNA or cells from individual organisms (mock communities), and technical replicates.
_In silico_ data and mock communities provide expected values for use in assessment but do not recapitulate the complexity of environmental samples or the error profile of real sequencing data.
Without an expected value technical replicates are only suitable for method comparison.
As part of this disseration I will introduce a new data type, mixtures of environmental samples, which provide the complexity of real data and an expected value for use in assessment.

## Dissertation Summary
In this dissertation I will present a framework for 16S rRNA marker-gene survey measurement assessment using a novel mixture dataset along with software to facilitate working with 16S rRNA marker-gene survey data.
The assessment framework was used to evaluate count table accuracy, differential abundance, and beta diversity analysis.
Count table and differential abundance estimate accuracy for different bioinformatic pipelines is assessed by comparing expected to observed values.
I will use the framework to assess how well bioinformatic pipelines and normalization methods minimize the impact of sequencing artifacts on beta-diversity analysis.
In addition to the assessment framework I developed Bioconductor R packages for working with 16S rRNA marker-gene survey data and 16S rRNA databases.
The assessment framework and software along with the assessment study results presented here will facilitate the development of improved computational methods and advance our understanding of the marker-gene survey measurement process.

### Measurement Process


### Count Table Assessment
__ADD AFTER REVISING__

### Diversity Assessment
Biases introduced during the marker-gene survey measurement process can impact downstream analysis such as beta-diversity.
Bioinformatic pipelines and normalization methods are often used to reduce biases introduced during the marker-gene survey measurement process [@Goodrich2014; @Kong2017].


Bioinformatic pipelines reduce bias by removing sequencing artifacts, such as single and multi-base pair variants, and chimeric sequences, from microbiome datasets.
If not accounted for, these artifacts may incorrectly be attributed as novel diversity in a sample.
Bioinformatic pipelines also use clustering or sequence inference techniques to group reads into biologically informative units.
Standard clustering methods include _de novo_ clustering based on pairwise sequence similarities [@Schloss2005] and closed reference clustering of reads against a reference database [@Edgar2011].
Open reference clustering is a combination of the two, first applying a closed reference approach, followed by _de novo_ clustering of reads that did not map to a reference [@Rideout2014].
Sequence inference methods use statistical models and algorithms to group sequences independent of sequence similarity but based on the probability that a lower abundant sequence is an artifact originating from more highly abundant sequence, independent of sequence similarity [@callahan2016DADA2; @Amir2017].
The resulting features, OTUs (operational taxonomic units) for clustering methods and SVs (sequence variants) for sequence inference methods, have different characteristics because the different methods vary in their ability to detect and remove errors, while retaining true biological sequences.


Rarefaction and numeric normalization methods account for differences in sample total abundances caused by uneven pooling of samples prior to sequencing and differences in sequencing run throughput.
Rarifying abundance data traces its origins to macro ecology, where counts for a unit (sample) are randomly subsampled to a user defined constant level [@Gotelli2001].
Although there are concerns about its statistical validity [@McMurdie2014],
rarefaction is currently the only normalization method for unweighted, presence-absence based, beta-diversity metrics [@Weiss2017].
For weighted, abundance based beta-diversity analyses, we can apply numeric normalization methods, such as total and cumulative sum scaling (TSS and CSS), where counts are divided by sample total abundance (TSS) or by the cumulative abundance (CSS) for a defined percentile [@Paulson2013].
CSS is one of the few normalization methods developed specifically for 16S rRNA marker-gene survey data.
Other normalization methods, including upper quartile (UQ), trimmed mean of M values (TMM) and relative log expression [@Robinson2010; @McCarthy2012], were initially developed for normalizing RNAseq and microarray data.
Many studies have found these methods useful in normalizing marker-gene survey data for differential abundance analysis, though it is unclear whether these techniques are also suitable for beta-diversity analysis.


Beta-diversity is calculated using a variety of metrics that can be grouped based on whether they account for phylogenetic distance and feature relative abundance.
The UniFrac metric, was developed specifically for marker-gene survey data and incorporates phylogenetic relatedness by comparing the branch lengths of features that are unique to two communities [@Hamady2010].
Unweighted UniFrac uses presence-absence information, whereas weighted UniFrac incorporates feature relative abundance.
Taxonomic metrics do not consider relationship between features.
The Bray-Curtis and Jaccard dissimilarity indices are examples of weighted and unweighted taxonomic metrics respectively, as they do not consider the phylogenetic relationship between features [@Bray1957; @Jaccard1912].
Beause these four groups of beta-diversity metrics measure different community characteristics,
they are not interchangeable should be evaluated in a complementary manner to gain maximal insight into community differences [@Anderson2011].


Previous studies have evaluated different bioinformatics pipelines [@Sinha2017] and normalization methods [@McMurdie2014; @Weiss2017] on beta-diversity analysis.
Yet, the ability of these pipelines to account for sequence quality and coverage, and how this affects diversity conclusions, remains unknown.
Here, we use novel dataset of stool samples from vaccine trial participants, collectec before and after exposure to the pathogen, and mixed following a two-sample titration mixture design.
We sequenced multiple technical PCR replicates, allowing us to evaluate (1) beta-diversity repeatability,
and hte ability to (2) distinguish between groups of samples with varying levels of similarity,
and (3) identify differences in beta-diversity between individuals and treatment.
Furthermore, the data was reproduced from across four runs with different sequencing error rates and library sizes,
enabling assessment of how each pipeline and method performs on datasets of varying quality.


### MetagenomeFeatures

A key step in 16S rRNA marker-gene surveys is comparing representative sequences  to a reference database for taxonomic classification or phylogenetic placement (Nguyen et al. 2016). There are numerous 16S rRNA reference databases of which Greengenes, RDP, and SILVA are arguably the most commonly used (DeSantis et al. 2006; Cole et al. 2014; Quast et al. 2012; McDonald et al. 2012).  Additionally, there are smaller system-specific databases such as HOMD for the human oral microbiome (Chen et al. 2010)(http://www.homd.org/) and soil reference database (Choi et al. 2017). System-specific databases can improve taxonomic assignments for microbial communities not well represented in the major databases (Rohwer et al. 2017).

16S rRNA databases differ in the number and diversity of sequences, the taxonomic classification system, and the inclusion of intermediate ranks (Balvočiūtė and Huson 2017, Table 1). Databases format their data differently and use sequence identification systems unique to their database, challenging membership and composition comparisons. For example, Yang et al. (2016) used the SILVA database to evaluate how different 16S rRNA variable regions impact phylogenetic analysis. Similarly, Martinez-Porchas et al. (2017) also used the SILVA database to evaluate sequence similarity between 16S rRNA gene conserved regions. Differences in database formatting present a significant barrier to performing the same analysis using multiple databases. Additionally, taxonomic assignments can be database-dependent, providing further justification for database comparisons (Pettengill and Rand 2017). To facilitate database comparisons RNACentral (http://rnacentral.org/) a resource combining non-coding RNA databases, provides unique identifiers for the sequences (The RNAcentral Consortium 2017).

We developed the R package _metagenomeFeatures_ for working with both 16S rRNA gene database and marker-gene survey feature data. _metagenomeFeatures_ provides a common data structure for working with 16S rRNA databases and marker-gene survey feature data. Additionally this package is the first step towards the development of a common data structure for use in analyzing metagenomic and marker-gene survey data using R bioconductor packages such as _phyloseq_ (McMurdie and Holmes 2013) and _metagenomeSeq_ (Paulson et al. 2013).


<!-- *  Finally, I will present the development of an R Bioconductor packages for working with 16S rRNA reference databases and marker-gene survey feature data. -->

<!-- Infrastructure
- Role of infrastructure in assessment
  * Computational infrastructure plays and important role in assessment.
  * Standardized data representation facilitates the comparison of measurement results obtained using different computational methods.
  * Additionally, a mature infrastructure includes data provenance, allowing for transparency in how the data was generated and computational reproducibility.
-->


<!-- 16S rRNA Gene
  * What is the 16S rRNA gene
  * Why is it used
  * Issues - copy number, within genome diversity, horizontal gene transfer.
    - 16S copy number https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0420-9
-->
