```{r include = FALSE}
knitr::opts_knit$set(root.dir = "mgtst_pub")
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
```

<!-- 
BMC Microbiome Submission https://microbiomejournal.biomedcentral.com/submission-guidelines/preparing-your-manuscript/methodology 

Rmarkdown with BMC template https://stackoverflow.com/questions/44373540/no-author-details-in-pdf-from-r-markdown-latex-template 
-->



## Abstract

## Introduction
```{r countIntroduction, child="mgtst_pub/Introduction.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
``` 


## Methods  
```{r countMethods, child="mgtst_pub/Methods.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
``` 

## Results
### Dataset characteristics  
```{r countSeqChar, child="mgtst_pub/results/seq_data_characteristics.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

### Titration Series Validation 
In order to use information from the unmixed samples to obtain expected count values for the titrations we first need to evaluate two assumptions about the mixed samples: 
1. The samples were mixed volumetrically in a $log_2$ dilution series according to the mixture design. 
2. The unmixed pre- and post-exposure samples have the same proportion of prokaryotic DNA. 
Exogenous DNA was spiked into the unmixed samples prior to mixing and quantified using qPCR to validate the samples were volumetrically mixed according to the mixture design. 
To evaluate the second assumption total prokaryotic DNA in the titrations samples was quantified using a qPCR assay targeting the 16S rRNA gene. 


#### Spike-in qPCR results   
```{r ercc, child="mgtst_pub/results/ercc_validation.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

#### Bacterial DNA Concentration 
```{r bacCon, child="mgtst_pub/results/bac_con_validation.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

#### Theta Estimates
```{r thetaEst, child="mgtst_pub/results/theta_estimate_results.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

### Measurement Assessment  
Next we assessed qualitative and quantitative nature of 16S rRNA measurement process using our two-sample titration dataset. 
For the qualitative assessment we analyzed the relative abundance of features only observed in the unmixed samples or titrations which are not expected to be observed given the titration experimental design. 
For the quantitative assessment we looked the the relative abundance and differential abundance log fold-change estimates. 

#### Qualitative Assessment

<!--
Proportion of titration and endpoint specific features that could not be explained by sampling alone. 
-->

```{r qualAnalysis, child="mgtst_pub/results/qualitative_assessment_results.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

#### Quantitative Assessment

<!--
Count Error Rate Results
-->
```{r quantAnalysis, child="mgtst_pub/results/relative_abundance_assessment_results.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

<!--
logFC Error Rate Results
-->
```{r logFCanalysis, child="mgtst_pub/results/logFC_assessment_results.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

## Discussion
```{r countDiscussion, child="mgtst_pub/Discussion.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
``` 


```{r supplemental, child="results/Supplemental.Rmd", warning=FALSE, message=FALSE, echo = FALSE, eval = FALSE}
```

