```{r include = FALSE}
knitr::opts_knit$set(root.dir = "mgtst_pub")
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
```

<!--
BMC Microbiome Submission https://microbiomejournal.biomedcentral.com/submission-guidelines/preparing-your-manuscript/methodology

Rmarkdown with BMC template https://stackoverflow.com/questions/44373540/no-author-details-in-pdf-from-r-markdown-latex-template
-->



## Abstract
```{r countAbstract, child="mgtst_pub/docs/Abstract.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

## Introduction
```{r countIntroduction, child="mgtst_pub/docs/Introduction.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```


## Methods
```{r countMethods, child="mgtst_pub/docs/Methods.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

## Results
### Dataset characteristics
```{r countSeqChar, child="mgtst_pub/results/seq_data_characteristics.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

### Titration Series Validation
To validate the two-sample titration dataset for use in abundance assessment we evaluated two assumptions about the titrations:
1. The samples were mixed volumetrically in a $log_2$ dilution series according to the mixture design.
2. The unmixed PRE and POST samples have the same proportion of prokaryotic DNA.
To validate the sample volumetric mixing exogenous DNA was spiked into the unmixed samples before mixing and quantified using qPCR .
To evaluate if the PRE and POST samples had the same proportion of prokaryotic DNA total prokaryotic DNA in the titrations samples was quantified using a qPCR assay targeting the 16S rRNA gene.


#### Spike-in qPCR results
```{r ercc, child="mgtst_pub/results/ercc_validation.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

#### Bacterial DNA Concentration
```{r bacCon, child="mgtst_pub/results/bac_con_validation.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

#### Theta Estimates
```{r thetaEst, child="mgtst_pub/results/theta_estimate_results.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

### Measurement Assessment
Next, we assessed the qualitative and quantitative nature of 16S rRNA measurement process using our two-sample titration dataset.
For the qualitative assessment, we analyzed the relative abundance of features only observed in the unmixed samples or titrations which are not expected given the titration experimental design.
The quantitative assessment evaluated relative and differential abundance estimates.

#### Qualitative Assessment

<!--
Proportion of titration and endpoint specific features that could not be explained by sampling alone.
-->

```{r qualAnalysis, child="mgtst_pub/results/qualitative_assessment_results.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

#### Quantitative Assessment

<!--
Count Error Rate Results
-->
```{r quantAnalysis, child="mgtst_pub/results/relative_abundance_assessment_results.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

<!--
logFC Error Rate Results
-->
```{r logFCanalysis, child="mgtst_pub/results/logFC_assessment_results.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```

## Discussion
```{r countDiscussion, child="mgtst_pub/docs/Discussion.Rmd", warning=FALSE, message=FALSE, echo = FALSE}
```


```{r supplemental, child="mgtst_pub/results/Supplemental.Rmd", warning=FALSE, message=FALSE, echo = FALSE, eval = FALSE}
```
