## Abstract
We developed the metagenomeFeatures R Bioconductor package along with annotation packages for the three primary 16S rRNA databases (Greengenes, RDP, and SILVA) to facilitate working with 16S rRNA sequence databases and marker-gene survey feature data. The metagenomeFeatures package defines two classes, `MgDb` for working with 16S rRNA sequence databases, and `mgFeatures` for working with marker-gene survey feature data. The associated annotation packages provide a consistent interface to the different 16S rRNA databases facilitating database comparison and exploration. The `mgFeatures` represents a crucial step in the development of a common data structure for working with 16S marker-gene survey data in R.

Availability: https://bioconductor.org/packages/release/bioc/html/metagenomeFeatures.html

## Introduction
16S rRNA marker-gene surveys have significantly advanced our understanding of the diversity and structure of prokaryotic communities present in ecosystems including the human gut, open ocean, and even the international space station [@lang2017-jk;@Thompson2017;@human_microbiome_project_consortium2012-au]. For a 16S rRNA marker-gene survey, the 16S rRNA gene is sequenced using a targeted assay. The raw sequence data is processed using a bioinformatic pipeline where the sequences are grouped into features, e.g., operational taxonomic units (OTUs) or sequence variants (SVs), yielding a set of representative sequences [@callahan2017-zv;@beiko2015-bn].

A critical step in 16S rRNA marker-gene surveys is comparing representative sequences to a reference database for taxonomic classification or phylogenetic placement [@nguyen2014tipp]. There are numerous 16S rRNA reference databases of which Greengenes, RDP, and SILVA are arguably the most commonly used [@desantis2006greengenes;@Cole2014;@quast2012silva;@McDonald2012-ro].  Additionally, there are smaller system-specific databases such as HOMD for the human oral microbiome [@chen2010-dw,http://www.homd.org/] and soil reference database [@choi2017-hv]. System-specific databases can improve taxonomic assignments for microbial communities not well represented in the major databases [@rohwer2017-fz].

16S rRNA databases differ in the number and diversity of sequences, the taxonomic classification system, and the inclusion of intermediate ranks [@balvociute2017-su](Table \ref@(tab:dbTable)). Databases format their data differently and use sequence identification systems unique to their database, challenging membership and composition comparisons. For example, Yang et al. @yang2016sensitivity used the SILVA database to evaluate how different 16S rRNA variable regions impact phylogenetic analysis. Similarly, Martinez-Porchas et al. @Martinez-Porchas2017-pi also used the SILVA database to evaluate sequence similarity between 16S rRNA gene conserved regions. Differences in database formatting present a significant barrier to performing the same analysis using multiple databases. Additionally, taxonomic assignments can be database-dependent, providing further justification for database comparisons [@pettengill2017-jw]. To facilitate database comparisons RNACentral (http://rnacentral.org/) a resource combining non-coding RNA databases, provides unique identifiers for the sequences [@the_rnacentral_consortium2017-ai].

We developed the R package metagenomeFeatures for working with both 16S rRNA gene database and marker-gene survey feature data. metagenomeFeatures provides a common data structure for working with the 16S rRNA databases and marker-gene survey feature data. Additionally, this package is the first step towards the development of a common data structure for use in analyzing metagenomic and marker-gene survey data using R packages such as _phyloseq_ [@mcmurdie2013-ij] and _metagenomeSeq_ [@paulson2013-ew].


\begin{table}[]
\centering
\caption{16S rRNA gene sequence databases with Bioconductor annotation packages we developed.}
\label{dbTable}
\begin{tabular}{@{}llll@{}}
\toprule
Database   & Version & Sequences & Taxonomic System \\ \midrule
Greengenes & 13.52   & 1,262,986 & NCBI             \\
RDP        & 11.5    & 3,356,809 & Bergey’s         \\
SILVA      & 128.1   & 1,922,213 & Bergey’s         \\ \bottomrule
\end{tabular}
\end{table}

## `MgDb` and `mgFeatures` Class Description
the _metagenomeFeatures_ package defines two data structures , `MgDb` for working with 16S rRNA databases, and `mgFeatures` for working with marker-gene survey feature data. There are three types of relevant information for both `MgDb` and `mgFeatures` class objects, (1) the sequences themselves, (2) sequence taxonomic lineage, and (3) a phylogenetic tree representing the evolutionary relationship between features. `MgDb` and `mgFeatures` data structures are both S4 object-oriented classes with slots for taxonomic, sequence, phylogenetic, and metadata.

As the 16S rRNA databases contain hundreds of thousands to millions of sequences, an SQLite database is used to store the taxonomic and sequence data. Using an SQLite database prevents the user from loading the full database into memory. The database connection is managed using the _RSQLite_ R package [@muller2017-ju]). The taxonomic data are accessed using the _dplyr_ and _dbplyr_ packages [@wickham2017-ed;@wickham2017-gv]. The DECIPHER package is used to format the sequence data as an SQLite database [@DECIPHER]. The phylo class, defined in a _APE_ R package, is used to define the tree slot [@paradis2004-do]. We developed Bioconductor annotation packages for commonly used databases, Greengenes, RDP, and SILVA (__Table REF__) [@Cole2014;@quast2012silva;@desantis2006-pa]. Along with database specific sequence identifiers RNAcentral identifiers are included in the SQLite table for inter-database comparisons.

`mgFeatures-class` is used for storing and working with marker-gene survey feature data.  Similar to the `MgDb-class`, the `mgFeatures-class` has four slots, for taxonomy, sequences,  phylogenetic tree, and metadata. As the number of features in a marker-gene survey dataset is significantly smaller than the number of sequences in a reference database, `mgFeatures-class` uses common Bioconductor data structures, DataFrames and DNAStringSets to define the taxonomic and sequence slots [@pages2008-xt;@pages2017-gh]. Similar to `MgDb-class`, a `phylo-class` object is used to define the tree slot. For both the `MgDb` and `mgFeatures` classes the tree slot is optional, and the metadata are stored as a list.

## Applications/ Vignettes
The metagenomeFeatures package includes a series of vignettes as example use cases for the metagenomeFeatures package and associated reference database annotation packages.
(1) Retrieving sequence and phylogenetic data for OTUs from closed-reference clustering.
(2) Exploring the diversity of a taxonomic group of interest.
(3) How to creating a `MgDb` class/ package for ecosystem specific database, e.g., oral microbiome.

The R command browseVignettes(“metagenomeFeatures”) provides a list of vignettes associated with the package and vignette(“x”) is used to view specific vignettes, where “x” is the vignette name.

To further demonstrate the utility of the package, the manuscript supplemental information uses metagenomeFeatures, greengenes13.5MgDb annotation package, and DECIPHER to evaluate the potential for species-level taxonomic classification using 16S rRNA V12 and V4 sequence data.

## Conclusions
The metagenomeFeatures package provides data structures and functions for working with 16S rRNA databases and marker-gene survey feature data. The data structure provided by the `MgDb-class` in conjunction with the shared sequence identifier system developed by RNACentral facilitates comparisons between 16S rRNA databases. The `mgFeatures-class` provides the groundwork for the development of a common data structure for working with metagenomic and marker-gene sequence data in R which will increase interoperability between R packages developed for working with metagenomic sequence data. Additionally, while the data structures were developed for 16S rRNA gene sequence data they can be used for any marker-gene sequence data without modification and can be extended to work with shotgun metagenomic sequence data and databases.
